### PR TITLE
fix: CI for py-ingestion-core comment workflow

### PR DESCRIPTION
**Pull request by akash-jain-10**
### Describe your changes :

Part 1 of the py-ingestion-publish CI revamp. This will comment on PR's opened with schema changes and let the Pull requestor notify of version bump to publish dev package in test.pypi.org. See #2239  and #2240 for reference.

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/openmetadata-ingestion-core-version-comment.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+name: OpenMetadata Ingestion Core Version Check Comment
+
+# read-write repo token
+# access to secrets
+on:
+  pull_request_target:
+    branches: [ main ]
+    types: [ opened ]
+    paths:
+    - 'catalog-rest-service/src/main/resources/json/**'
```
Comment (pmbrull) [inline .github/workflows/openmetadata-ingestion-core-version-comment.yml:21]: If I remember correctly from previous tests the filter does not quite work for the pull_request_target. We might need to check https://github.com/open-metadata/OpenMetadata/issues/2185 and use https://github.com/dorny/paths-filter to validate the paths on the checked out code!

Comment (akash-jain-10) [reply .github/workflows/openmetadata-ingestion-core-version-comment.yml:21]: As per documentation, this should work with `pull_request_target`. 
Ref - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Comment (pmbrull) [reply .github/workflows/openmetadata-ingestion-core-version-comment.yml:21]: Then I cannot remember why I started looking into that action 😂  I must be mixing something here. LGTM!

**Review by pmbrull — APPROVED** [review]
Great thanks for this 🚀

**Pull request merged**